br/pkg/streamhelper/advancer_cliext.go

```diff
@@ -127,13 +154,22 @@ func (t AdvancerExt) startListen(ctx context.Context, rev int64, ch chan<- TaskE
 		}
 		return true
 	}
+
+	// inner function def
 	collectRemaining := func() {
 		log.Info("Start collecting remaining events in the channel.", zap.String("category", "log backup advancer"),
-			zap.Int("remained", len(c)))
+			zap.Int("remained", len(taskCh)))
 		defer log.Info("Finish collecting remaining events in the channel.", zap.String("category", "log backup advancer"))
 		for {
 			select {
-			case resp, ok := <-c:
+			case resp, ok := <-taskCh:
+				if !ok {
+					return
+				}
+				if !handleResponse(resp) {
+					return
+				}
+			case resp, ok := <-pauseCh:
```

Comment:
We may want to "merge" all remaining events from both `taskCh` and `pauseCh`, so I think we'd better return once both `taskCh` and `pauseCh` are closed.
Tips: a `nil` channel in go will block forever, so you may consume all data in the two channels via replacing the `!ok` branch with:
```go
// ...
if !ok {
  taskCh = nil
}
// ...
```